### ISSUE TITLE
Exclusive `Iter.range()`?

### ISSUE THREAD

**Issue by rvanasa** [dfinity/new-motoko-base#31]
Do we want to change the behavior of `Iter.range()` to exclude the upper bound? For example, `Iter.toArray(Iter.range(0, 3))` would return `[0, 1, 2]` instead of `[0, 1, 2, 3]`.

While an exclusive upper bound is the expected behavior for a zero-indexed language, making this change could silently break people's code during migration to the new base library. Opening this issue for a place to discuss whether we should do this.

**Comment by michael-weigelt** [comment]
As a new user, +1.

If a base library rework is in progress anyway, that could include a whole bunch of breaking changes at once. The problem with this one is that it's a silent one. But people who upgrade their dependencies to a new major version must be prepared to read the release notes and fix the breaking changes...

**Comment by rvanasa** [comment]
> The problem with this one is that it's a silent one.

I totally agree with this. We may have found a good solution in the form of replacing `Iter.range()` with type-specific `Int.range()`, `Nat.range()`, etc. with the new functionality. See #54 for more details.
